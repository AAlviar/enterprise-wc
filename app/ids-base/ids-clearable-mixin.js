import IdsClearableMixin from '../../src/ids-base/ids-clearable-mixin';
