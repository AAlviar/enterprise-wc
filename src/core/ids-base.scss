/* Sass scss styles. Keep Base very small */
@import '../../node_modules/ids-identity/dist/theme-new/tokens/web/theme-new-mixins';
@import '../mixins/sass/ids-general-mixins';

/* Global variables (might want these to become tokens) */
$breakpoint-xxl: 1440px;
$breakpoint-xl: 1280px;
$breakpoint-lg: 1024px;
$breakpoint-md: 840px;
$breakpoint-sm: 600px;
$breakpoint-xs: 360px;

$transition-length: 200ms;
$transition-ease: cubic-bezier(0.4, 0, 0.6, 1);

// Field Sizes
$input-size-xs: 75px;
$input-size-sm: 150px;
$input-size-mm: 225px;
$input-size-md: 300px ;
$input-size-lg: 400px;
$input-size-full: 100%;

// Field Heights
$input-field-height-xs: 28px;
$input-field-height-sm: 30px;
$input-field-height-md: 38px;
$input-field-height-lg: 48px;

// Sometimes we need to "fade" white into transparency, so we need a
// completely-transparent "white" color to use as one end of a gradient.
$transparent-white: rgba(255 255 255 / 0);

// Other semi-transparent colors (temporary until we move these to design tokens)
// See infor-design/design-system#490
$ids-slider-thumb-color-light: rgba(0 114 237 / 0.1);
$ids-slider-thumb-readonly-shadow-color-light: rgba(119 119 124 / 0.1);
$ids-slider-thumb-color-dark: rgba(0 114 237 / 0.1);
$ids-slider-thumb-readonly-shadow-color-dark: rgba(151 151 155 / 0.1);
$ids-slider-thumb-color-contrast: rgba(0 74 153 / 0.1);
$ids-slider-thumb-readonly-shadow-color-contrast: rgba(71 71 76 / 0.1);
$ids-slider-thumb-hover-shadow-light: rgba(0 0 0 / 0.4);
$ids-slider-thumb-hover-shadow-dark: rgba(255 255 255 / 0.4);
$ids-slider-thumb-hover-shadow-contrast: rgba(0 0 0 / 0.4);
