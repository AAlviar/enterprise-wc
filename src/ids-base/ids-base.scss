/* Ids Tokens - Exposed as Css Variables - Root/Global Tokens */
:root {
  /* Ids Color Pallette */
  --ids-theme-color-palette-graphite-20: #D3D3D3;
  --ids-theme-color-palette-graphite-30: #B1B1B1;
  --ids-theme-color-palette-white: #ffffff;
  --ids-theme-color-status-caution: #FFD726;
  --ids-theme-color-status-danger: #e84f4f;
  --ids-theme-color-status-error: #DA1217;
  --ids-theme-color-status-success: #2AC371;
  --ids-theme-color-status-warning: #ff9426;

  --ids-theme-theme-color-font-base: #1a1a1a;

  /* Ids Font Sizes */
  --ids-theme-size-font-px-14: 14px; 
}

/* Ids Font Charactistics */
@mixin font-sans {
  font-family: "source sans pro", helvetica, arial, san-serif;
}

@mixin antialiased {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

@mixin text-bold {
  font-style: bold;
}

/* Ids Font Sizes */
@mixin text-px-14 {
  font-size: var(--ids-theme-size-font-px-14);
}

/* Ids Font Colors */
@mixin text-base {
  color: var(--ids-theme-theme-color-font-base);
}

/* Background Colors */
@mixin bg-graphite-20 {
  background-color: var(--ids-theme-color-palette-graphite-20);
}

@mixin bg-white {
  background-color: var(--ids-theme-color-palette-white);
}

/* Borders */
@mixin border {
  border: 1px solid;
}

@mixin rounded-md {
  border-radius: 15px;
}

@mixin border-graphite-20 {
  border-color: var(--ids-theme-color-palette-graphite-20);
}

@mixin border-graphite-30 {
  border-color: var(--ids-theme-color-palette-graphite-30);
}