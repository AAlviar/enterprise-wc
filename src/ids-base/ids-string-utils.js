/**
 * Convert a string in presumed kebab case to camel case
 * @param  {string} str [description]
 * @returns {string} The return string
 */
export function camelCase(str) {
  return (str.slice(0, 1).toLowerCase() + str.slice(1))
    .replace(/([-_ ]){1,}/g, ' ')
    .split(/[-_ ]/)
    .reduce((cur, acc) => cur + acc[0].toUpperCase() + acc.substring(1));
}

/**
 * Removes all duplicate characters from a string and returns another string
 * containing ALL unique characters.  Useful for construction of REGEX objects
 * with characters from an input field, etc.
 * @param {string} str The string to process
 * @returns {string} The processed string
 */
export function removeDuplicates(str) {
  return str
    .split('')
    .filter((item, pos, self) => self.indexOf(item) === pos)
    .join('');
}

/**
 * Convert a string value into a boolean
 * @param {string|boolean|any} val string value from the component property
 * @returns {boolean} The return boolean
 */
export function stringToBool(val) {
  if (typeof val === 'string' && val.toLowerCase() === 'false') {
    return false;
  }
  return val !== null && (val === true || (typeof val === 'string' && val !== 'false'));
}

/**
 * Convert a string value into a number
 * @param {string|number|any} val string value from the component property
 * @returns {number} The return boolean
 */
export function stringToNumber(val) {
  const v = val?.toString() * 1; // Converting String to Number
  return !isNaN(v) ? v : 0; // eslint-disable-line
}

/**
 * Inject template variables in a string
 * @param {string} str The string to inject into
 * @param {string} obj The string to inject into
 * @returns {obj} The dataset row / item
 */
export function injectTemplate(str, obj) {
  return str.replace(/\${(.*?)}/g, (_x, g) => obj[g]);
}

/**
 * combines classes and considers truthy/falsy +
 * doesn't pollute the attribs/DOM without
 * any fuss
 *
 * @param  {...any} classes classes/expressions
 * @returns {string} ` class="c1 c2..."` || ""
 */
export function buildClassAttrib(...classes) {
  /* istanbul ignore next */
  const classAttrib = classes.reduce((attribStr = '', c) => {
    /* istanbul ignore else */
    if (attribStr && c) { return `${attribStr} ${c}`; }

    /* istanbul ignore else */
    if (!attribStr && c) { return c; }
    return attribStr;
  }, '');

  /* istanbul ignore next */
  return !classAttrib ? '' : ` class=${classAttrib}`;
}

/**
 * Ids String parsing/processing utilities
 */
export const IdsStringUtils = {
  camelCase,
  injectTemplate,
  stringToBool,
  stringToNumber,
  removeDuplicates,
  buildClassAttrib
};

export default IdsStringUtils;
export { IdsStringUtils as stringUtils };
