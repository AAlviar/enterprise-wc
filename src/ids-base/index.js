export { default as IdsThemeMixin } from './ids-theme-mixin';
export { default as IdsEventsMixin } from './ids-events-mixin';
export { default as IdsKeyboardMixin } from './ids-keyboard-mixin';
