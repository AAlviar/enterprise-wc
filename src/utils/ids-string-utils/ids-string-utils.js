/**
 * Convert a string in presumed kebab case to camel case
 * @param  {string} str [description]
 * @returns {string} The return string
 */
export function camelCase(str) {
  return (str.slice(0, 1).toLowerCase() + str.slice(1))
    .replace(/([-_ ]){1,}/g, ' ')
    .split(/[-_ ]/)
    // eslint-disable-next-line no-unsafe-optional-chaining
    .reduce((cur, acc) => cur + acc[0]?.toUpperCase() + acc.substring(1));
}

/**
 * Removes all duplicate characters from a string and returns another string
 * containing ALL unique characters.  Useful for construction of REGEX objects
 * with characters from an input field, etc.
 * @param {string} str The string to process
 * @returns {string} The processed string
 */
export function removeDuplicates(str) {
  return str
    .split('')
    .filter((item, pos, self) => self.indexOf(item) === pos)
    .join('');
}

/**
 * Convert an attribute string into a boolean representation
 * @param {string|boolean|any} val string value from the component attribute
 * @returns {boolean} The return boolean
 */
export function stringToBool(val) {
  if (typeof val === 'string' && val.toLowerCase() === 'false') {
    return false;
  }
  return val !== null && (val === true || (typeof val === 'string' && val !== 'false'));
}

/**
 * Converts an attribute string into a number
 * @param {string|number|any} val string value from the component attribute
 * @returns {number} The return boolean
 */
export function stringToNumber(val) {
  // eslint-disable-next-line no-unsafe-optional-chaining
  const v = val?.toString() * 1; // Converting String to Number
  return !isNaN(v) ? v : 0; // eslint-disable-line
}

/**
 * Inject template variables in a string
 * @param {string} str The string to inject into
 * @param {string} obj The string to inject into
 * @returns {obj} The dataset row / item
 */
export function injectTemplate(str, obj) {
  return str.replace(/\${(.*?)}/g, (_x, g) => obj[g]);
}

/**
 * Combines classes and considers truthy/falsy +
 * doesn't pollute the attribs/DOM
 * @param  {...any} classes classes/expressions
 * @returns {string} ` class="c1 c2..."` || ""
 */
export function buildClassAttrib(...classes) {
  const classAttrib = classes.reduce((attribStr, c) => {
    if (attribStr && c) { return `${attribStr} ${c}`; }

    if (!attribStr && c) { return c; }
    return attribStr;
  }, '');

  return !classAttrib ? '' : ` class="${classAttrib}"`;
}

/**
 * Checks a keycode value and determines if it belongs to a printable character
 * @private
 * @param {number} e the event to inspect
 * @returns {boolean} Returns true if the key is a printable one.
 */
export function isPrintable(e) {
  const controlKeys = ['Alt', 'Shift', 'Control', 'Meta', 'CapsLock', 'Enter', 'Escape', 'Tab'];
  if (controlKeys.indexOf(e.key) > -1 || e.key.indexOf('Arrow') > -1) {
    return false;
  }

  if ((e.altKey && (e.keyCode === 38)) || (e.keyCode > 111 && e.keyCode < 124)) {
    return false;
  }
  return true;
}
