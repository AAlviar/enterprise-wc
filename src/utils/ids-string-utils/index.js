export { default as IdsStringUtils } from './ids-string-utils';
