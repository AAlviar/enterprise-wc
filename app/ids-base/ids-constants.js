import { attributes } from '../../src/ids-base/ids-attributes'; //eslint-disable-line
