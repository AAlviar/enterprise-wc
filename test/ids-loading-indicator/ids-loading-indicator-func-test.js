/**
 * @jest-environment jsdom
 */
import { stringUtils } from '../../src/ids-base/ids-string-utils';
import processAnimFrame from '../helpers/process-anim-frame';
import IdsLoadingIndicator from '../../src/ids-loading-indicator';

const expectElemFlagBehavior = (elem, attribute, initialValue = false) => {
  const camelCasedAttrib = stringUtils.camelCase(attribute);

  expect(elem[camelCasedAttrib]).toEqual(initialValue);

  elem[camelCasedAttrib] = true;
  expect(elem[camelCasedAttrib]).toEqual(true);

  elem.removeAttribute(attribute);
  expect(elem[camelCasedAttrib]).toEqual(false);

  elem.setAttribute(attribute, 'true');
  expect(elem.hasAttribute(attribute)).toEqual(true);

  elem.removeAttribute(attribute);
  expect(elem[camelCasedAttrib]).toEqual(false);

  elem.setAttribute(attribute, '');
  expect(elem[camelCasedAttrib]).toEqual(true);

  elem[camelCasedAttrib] = false;
  expect(elem.hasAttribute(attribute)).toEqual(false);
};

describe('IdsLoadingIndicator Component', () => {
  let elem;

  const createElemViaTemplate = async (innerHTML) => {
    elem?.remove?.();

    const template = document.createElement('template');
    template.innerHTML = innerHTML;
    elem = template.content.childNodes[0];
    document.body.appendChild(elem);

    await processAnimFrame();

    return elem;
  };

  afterEach(async () => {
    elem.remove();
  });

  it('renders circular/indeterminate (default) with no errors', async () => {
    const errors = jest.spyOn(global.console, 'error');

    elem = await createElemViaTemplate(
      '<ids-loading-indicator />'
    );

    expect(document.querySelectorAll('ids-loading-indicator').length).toEqual(1);
    expect(errors).not.toHaveBeenCalled();

    expect(elem.shadowRoot.innerHTML).toMatchSnapshot();
  });

  it('renders linear/indeterminate indicator without error', async () => {
    const errors = jest.spyOn(global.console, 'error');

    elem = await createElemViaTemplate(
      '<ids-loading-indicator linaer />'
    );

    expect(document.querySelectorAll('ids-loading-indicator').length).toEqual(1);
    expect(errors).not.toHaveBeenCalled();

    expect(elem.shadowRoot.innerHTML).toMatchSnapshot();
  });

  it('renders linear/determinate indicator without error', async () => {
    const errors = jest.spyOn(global.console, 'error');

    elem = await createElemViaTemplate(
      '<ids-loading-indicator linear />'
    );

    expect(document.querySelectorAll('ids-loading-indicator').length).toEqual(1);
    expect(errors).not.toHaveBeenCalled();

    expect(elem.shadowRoot.innerHTML).toMatchSnapshot();
  });

  it('renders sticky/indeterminate indicator without error', async () => {
    const errors = jest.spyOn(global.console, 'error');

    elem = await createElemViaTemplate(
      '<ids-loading-indicator sticky />'
    );

    expect(document.querySelectorAll('ids-loading-indicator').length).toEqual(1);
    expect(errors).not.toHaveBeenCalled();

    expect(elem.shadowRoot.innerHTML).toMatchSnapshot();
  });

  it('sets and gets the progress attribute reliably', async () => {
    elem = await createElemViaTemplate(
      '<ids-loading-indicator progress="30" />'
    );

    expect(elem.progress).toEqual(30);

    elem.progress = '90';
    expect(elem.progress).toEqual(90);

    elem.progress = null;
    expect(elem.progress).toEqual(undefined);
  });

  it('sets and gets the sticky attribute reliably', async () => {
    elem = await createElemViaTemplate(
      '<ids-loading-indicator />'
    );

    expectElemFlagBehavior(elem, 'sticky');
  });

  it('sets and gets the linear attribute reliably', async () => {
    elem = await createElemViaTemplate(
      '<ids-loading-indicator />'
    );

    expectElemFlagBehavior(elem, 'linear');
  });

  it('sets and gets the inline attribute reliably', async () => {
    elem = await createElemViaTemplate(
      '<ids-loading-indicator />'
    );

    expectElemFlagBehavior(elem, 'inline');
  });

  it('sets the sticky attribute after setting the linear but becomes only sticky', async () => {
    elem = await createElemViaTemplate('<ids-loading-indicator linear />');

    elem.sticky = true;

    expect(elem.hasAttribute('linear')).toEqual(false);
  });

  it('calls type getter reliably based on flags set', async () => {
    elem = await createElemViaTemplate('<ids-loading-indicator linear />');
    expect(elem.type).toEqual('linear');

    elem.setAttribute('sticky', true);
    expect(elem.type).toEqual('sticky');

    elem.removeAttribute('sticky');
    expect(elem.type).toEqual('circular');
  });

  it('sets and gets the percentage-visible attribute reliably', async () => {
    elem = await createElemViaTemplate(
      '<ids-loading-indicator />'
    );

    expectElemFlagBehavior(elem, 'percentage-visible');
  });

  it('supports setting mode', async () => {
    elem = await createElemViaTemplate(
      '<ids-loading-indicator />'
    );
    elem.mode = 'dark';
    expect(elem.getAttribute('mode')).toEqual('dark');
  });

  it('supports setting version', () => {
    elem.version = 'classic';
    expect(elem.getAttribute('version')).toEqual('classic');
  });
});
