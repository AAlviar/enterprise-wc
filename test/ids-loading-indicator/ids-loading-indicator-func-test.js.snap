// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdsLoadingIndicator Component renders correctly 1`] = `
"<style nonce=\\"0a59a005\\">.ids-loading-indicator { background-color: transparent; }</style><svg viewBox=\\"0 0 100 100\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"circular-indicator indeterminate\\">
        <circle cx=\\"50\\" cy=\\"50\\" r=\\"45\\" stroke-width=\\"3\\" class=\\"overall\\"></circle>
        <circle cx=\\"50\\" cy=\\"50\\" r=\\"45\\" stroke-width=\\"6\\" class=\\"progress\\"></circle>
      </svg>
      "
`;
